extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisRecordProcessor.scala

```diff
@@ -135,18 +110,17 @@ private[kinesis] class KinesisRecordProcessor[T](
    */
   override def shutdown(checkpointer: IRecordProcessorCheckpointer, reason: ShutdownReason) {
     logInfo(s"Shutdown:  Shutting down workerId $workerId with reason $reason")
```

Comment:
Are you sure that this shutdown gets called when the receiver is stopped? If it is not, then this will not do the final checkpointing on receiver shutdown.